content/blog/2021-07-13-hyperparam-tuning.md

```diff
@@ -0,0 +1,294 @@
+---
+title: Tuning Hyperparameters with Reproducible Experiments
+date: 2021-07-13
+description: |
+  It's easy to lose track of which changes gave you the best result when you start exploring multiple model architectures. Tracking the changes in your hyperparameter values, along with code and data changes, will help you build a more efficient model by giving you an exact reproduction of the conditions that made the model better.
+
+descriptionLong: |
+  Since hyperparameter tuning is a common task in ML, it's important to know the values and changes that lead to the model you want to push to production. When you add reproducibility to your project with DVC, you'll be able to track all of the changes you make that give you the optimum model.
+
+picture: 2021-07-13/tuning-hyperparams.png
+pictureComment: Tuning Hyperparameters with Reproducible Experiments
+author: milecia_mcgregor
+commentsUrl: https://discuss.dvc.org/t/utilizing-custom-git-references-in-dvc/727
+tags:
+  - MLOps
+  - Git
+  - Experiments
+  - Reproducibility
+---
+
+## Intro
+
+When you're starting to build a new machine learning model and you're deciding
+on the model architecture, there are a number of issues that arise. You have to
+monitor every code change you make, note any differences in your data, and keep
+up with hyperparameter value updates.
+
+This means it's important for you to be able to track all of these changes so
+you can reproduce your experiments without wondering which changes gave you the
+best model. Having reproducibility in your training gives you the ability to go
+back to any point in your process to see which changes gave you the best
+results.
+
+In this tutorial, we're going to go through an example of hyperparameter tuning
+with reproducibility using DVC. You can add this to any existing project you're
+working on or start from a fresh project.
+
+## Background on Hyperparameters
+
+Before jumping straight into training and experiments, let's briefly go over
+some background on hyperparameters. Hyperparameters are the values that define
+your model. This includes things like the number of layers in a neural network
+or the learning rate for gradient descent.
+
+These parameters are different from model parameters because we can't get them
```

Comment:
Here's a pretty quick article on the difference. 🙂  https://towardsdatascience.com/model-parameters-and-hyperparameters-in-machine-learning-what-is-the-difference-702d30970f6

Reply:
Thanks! Still "values that define your model" (definition of hyperparams in line 41) sounds like "model parameters" to me. Or at least both model params and hyperparams define aspects of the model, so that's still a somewhat confusing definition from my PoV. But if you think post readers won't have this confusion then 🆗 

> Tip: from https://machinelearningmastery.com/difference-between-a-parameter-and-a-hyperparameter/ it seems hyperparams are **external** to the model. Maybe that's a helpful terminology distinction (internal vs external).